### PR TITLE
Improvements to reloading [RDY]

### LINKED ISSUE #17211: Pipe rifles & revolvers not ejecting empty casings
... until you reload a second round. This means you never get spent casing from pipe rifles since they can't hold more than a single round.

Example: fire all six shots from a revolver till it's empty, reload once - no ejection. Reload second time, get all six casings.

If you keep loading and firing only a single cartridge at a time, you can build up a huge backlog of brass that pours out like a deluge the moment you load two in a row.

### PR DESCRIPTION
**Pull request by mugling**
This PR moves the `item::pick_reload_ammo` to `player::select_ammo` and separates the UI code from the item search code. A lot of the line changes relate to this, especially aede4f4 and 6262f03. This allows the same menu to be used by other tasks which is exploited to allow bandoliers to have similar semantics to guns and tools (can be unloaded and reloaded).

`iuse::quiver` is deprecated and replaced with `iuse_actor::bandolier`. It is possible to set a `draw_cost` for bandoliers (fdb7feb) which is used to further reduce the moves cost of archery. Along with dropping the spurious penalty (first suggested in #17272) and earlier changes in #17274 and  #17453 provides further mitigation for #16545. I want to look at item handling costs again but this beyond the scope of this PR.

Also provides fix for casing ejection closing #17211 and extends `item::info` to display casings that have yet to be ejected (8df1e4d)

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -11245,6 +11245,8 @@ bool game::unload( item &it )
         return false;
     }
 
+    target->eject_casings( g->u.pos() );
```
Comment (BevapDin) [inline src/game.cpp]: The `g->` is not needed here, `u.pos()` should suffice.

File: data/json/items/armor.json [hunk 2]
```diff
@@ -6633,8 +6643,13 @@
...
         "coverage" : 15,
         "material_thickness" : 1,
-        "flags" : ["QUIVER_60", "BELTED", "VARSIZE", "OVERSIZE"],
-        "use_action" : "QUIVER"
+        "flags" : [ "BELTED", "VARSIZE", "OVERSIZE" ],
+        "use_action" : {
+            "type": "bandolier",
+            "capacity": 60,
+            "ammo": [ "arrow" ],
+            "draw_cost": 5
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: No need to penalize draw cost, the encumbrance penalty is enough.

Comment (mugling) [inline data/json/items/armor.json]: As in set it equivalent to `quiver`?

**Comment by Coolthulhu** [comment]
Not necessary here, but:
Now that all reload/unload checks include `is_gun() || is_tool() || is_bandolier()`, it would be a good idea to have something along the lines of `is_reloadable()`.

**Pull request merged**